src/utils/fastmath.h

```diff
@@ -65,4 +65,10 @@ inline float FastLog(const float a) {
   return 0.6931471805599453f * FastLog2(a);
 }
 
+// Fast approximate exp(x).
+inline float FastExp(const float a) {
+  return (a < 0) ? 1.0f / FastPow2(-1.442695040f * a)
```

Comment:
Maybe a different approximation would be better here.

Reply:
That's certainly a possibility. With the latest changes, the `FastLog2` call has been replaced by `FastExp` (which uses `FastLog2` internally), so I don't expect worse performance. Please let me know if you have any further doubts.

It might for example be worth investigating if there is much difference (in terms of elo) w.r.t. `std::exp`, but I fear we might be repeating the investigations in #580.